src/Symfony/Component/Security/Core/Authentication/Provider/SimpleAuthenticationProvider.php

```diff
@@ -44,6 +45,10 @@ public function authenticate(TokenInterface $token)
             throw new AuthenticationException('Simple authenticator failed to return an authenticated token.');
         }
 
+        if ($authToken instanceof AnonymousToken) {
```

Comment:
the right check would be `!$user instanceof UserInterface`, to cover all cases where calling the user checker is not possible.

Reply:
done